src/towncrier/_vcs.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) towncrier contributors, 2025
+# See LICENSE for details.
+
+from __future__ import annotations
+
+import os
+
+from typing import Container, Protocol
+
+
+class VCSMod(Protocol):
+    def get_default_compare_branch(self, branches: Container[str]) -> str | None: ...
+    def remove_files(self, fragment_filenames: list[str]) -> None: ...
+    def stage_newsfile(self, directory: str, filename: str) -> None: ...
+    def get_remote_branches(self, base_directory: str) -> list[str]: ...
+
+    def list_changed_files_compared_to_branch(
+        self, base_directory: str, compare_with: str, include_staged: bool
+    ) -> list[str]: ...
+
+
+def _get_mod(base_directory: str) -> VCSMod:
+    base_directory = os.path.abspath(base_directory)
+    if os.path.exists(os.path.join(base_directory, ".git")):
+        from . import _git
+
+        return _git
+    elif os.path.exists(os.path.join(base_directory, ".hg")):
+        from . import _hg
+
+        hg: VCSMod = _hg
+
+        return hg
+    else:
+        # No VCS was found in the current directory
+        # We will try our luck in the parent directory.
+        parent = os.path.dirname(base_directory)
+        if parent == base_directory:
+            # We reached the fs root, abandoning
+            from . import _novcs
+
+            return _novcs
+
+        return _get_mod(parent)
+
+
+def get_default_compare_branch(
```

Comment:
I like the way git and Mercurial was implemeted and I think that for this PR the current changes are ok.
No special need to extra work here.

---------

I think that for a future PR, it would be nice to have HG and GIT as a class that implement the `VCS` protocol, so that we don't have to create this type of code